src/main/kotlin/org/rust/lang/core/grammar/rust.bnf

```diff
@@ -805,6 +804,8 @@ left view_path_part ::= '::' (SUPER | IDENTIFIER) {
 //
 expr ::=
 
+// https://doc.rust-lang.org/reference.html#operator-precedence
+// https://github.com/JetBrains/Grammar-Kit/blob/master/HOWTO.md#24-compact-expression-parsing-with-priorities
```

Comment:
The nice description of the algorithm is here: http://journal.stuffwithstuff.com/2011/03/19/pratt-parsers-expression-parsing-made-easy/